pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -321,102 +351,77 @@ func (ormClient *ORMClient) LocateOwnerPaths(ownedObj *unstructured.Unstructured
-			continue
-		}
-		re := regexp.MustCompile(`(\[)|(\]\.)`)
-		parsedDestPath := re.ReplaceAllString(destPath, ".")
-		fields := strings.Split(parsedDestPath, ".")
-		if len(fields) < 2 {
-			return fmt.Errorf("failed to update %v to CR %s for %s in namespace %s: '%s' is invalid path", newValue, operatorRes, componentKey, resourceNamespace, destPath)
-		}
-		origCRValue, _, err := util.NestedField(operatorCR, resourceMappingDestPath, destPath)
-		if err != nil {
-			return fmt.Errorf("failed to update %v to CR %s '%s' for %s in namespace %s: %v", newValue, operatorRes, destPath, componentKey, resourceNamespace, err)
-		}
-		err = util.SetNestedField(operatorCR.Object, newValue, fields[1:]...)
-		if err != nil {
-			return fmt.Errorf("failed to update %v to CR %s '%s' for %s in namespace %s: %v", newValue, operatorRes, destPath, componentKey, resourceNamespace, err)
-		}
-		_, err = dynResourceClient.Update(context.TODO(), operatorCR, metav1.UpdateOptions{})
-		if err != nil {
-			return fmt.Errorf("failed to update %v to CR %s '%s' for %s in namespace %s: %v", newValue, operatorRes, destPath, componentKey, resourceNamespace, err)
+	sourceResKind := updatedControllerObj.GetKind()
+	sourceResName := updatedControllerObj.GetName()
+	sourceRes := sourceResKind + "/" + sourceResName
+	for ownedPath, resourcePaths := range ownerResources.OwnerResourcesMap {
+		for _, resourcePath := range resourcePaths {
+			// Retrieve the owner object and path
+			ownerPath := resourcePath.Path
+			ownerObj := resourcePath.ObjectReference
+			ownerResKind := ownerObj.Kind
+			ownerResName := ownerObj.Name
+			ownerRes := ownerResKind + "/" + ownerResName
+			ownerResNamespace := ownerObj.Namespace
+			// ownerResources might have source/owned resource kind with their resource paths if it cannot find the owner resource mapping from ORM.
+			// In that case we cannot perform this update operation on source/owned resource kind without owner resource found
+			if ownerResKind != updatedControllerObj.GetKind() {
+				glog.Infof("Update owner %s/%s resources found for source %s/%s",
+					ownerResKind, ownerResName,
+					sourceResKind, sourceResName)
+				ownerCR, err := kubernetes.Toolbox.GetResourceWithObjectReference(ownerObj)
+				if err != nil {
+					return fmt.Errorf("failed to get orm v2 owner CR for owner object %s in namespace %s: %v", ownerRes, ownerResNamespace, err)
+				}
+				// get the new resource value from the source obj
+				newCRValue, found, err := ormutils.NestedField(updatedControllerObj, resourceMappingSrcPath, ownedPath)
+				if err != nil || !found {
+					return fmt.Errorf("failed to get value for source/owned resource %s for path '%s' in updatedControllerObj, error: %v", sourceRes, ownerPath, err)
+				}
+				// get the original resource value from the owner obj
+				origCRValue, found, err := ormutils.NestedField(ownerCR, resourceMappingDestPath, ownerPath)
+				if err != nil || !found {
```

Comment:
This is incorrect. resourceMappingDestPath is a pointer into the v1 ORM to get the onwer resource path not in the controller Object. This
The resource value you want to get from the 'updatedControllerObj' is at the ownerPath ('resourcePath') location

Reply:
same comment from above